managed/resource_api_key.go

```diff
@@ -0,0 +1,334 @@
+	if err != nil {
+		errMsg := getErrorMessage(response, err)
+		return apiKey, false, errMsg
+	}
+
+	apiKey.AccountID.Value = accountId
+	apiKey.ProjectID.Value = projectId
+	apiKey.ApiKeyID.Value = apiKeyId
+	apiKey.Name.Value = apiKeyResp.Data.Spec.Name
+	apiKey.Duration.Value = duration
+	apiKey.Unit.Value = unit
+	apiKey.ApiKey.Value = key
+	description := apiKeyResp.Data.Spec.GetDescription()
+	if description != "" {
+		apiKey.Description.Value = description
+	}
+	apiKey.RoleName.Value = apiKeyResp.Data.Info.Role.Info.GetDisplayName()
+	apiKey.Status.Value = string(apiKeyResp.Data.Info.Status)
+	apiKey.Issuer.Value = apiKeyResp.Data.Info.Issuer
+	apiKey.LastUsed.Value = apiKeyResp.Data.Info.GetLastUsedTime()
+	apiKey.ExpiryTime.Value = apiKeyResp.Data.Info.ExpiryTime
+	apiKey.CreatedAt.Value = apiKeyResp.Data.Info.Metadata.GetCreatedOn()
+
+	return apiKey, true, ""
+}
+
+// Read API Key
+func (r resourceApiKey) Read(ctx context.Context, req tfsdk.ReadResourceRequest, resp *tfsdk.ReadResourceResponse) {
+	var state ApiKey
+	getIDsFromApiKeyState(ctx, req.State, &state)
+
+	tflog.Debug(ctx, "API Key Read: API Key name and role from state", map[string]interface{}{
+		"Name":     state.Name,
+		"RoleName": state.RoleName})
+
+	apiKey, readOK, message := resourceApiKeyRead(state.AccountID.Value, state.ProjectID.Value, state.ApiKeyID.Value, state.Duration.Value, state.Unit.Value, state.ApiKey.Value, r.p.client)
+	if !readOK {
+		resp.Diagnostics.AddError("Unable to read the state of the API Key ", message)
+		return
+	}
+	tflog.Debug(ctx, "API Key Read: API Key on read from API server", map[string]interface{}{
+		"API Key": apiKey})
+
+	diags := resp.State.Set(ctx, &apiKey)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update API Key
+func (r resourceApiKey) Update(ctx context.Context, req tfsdk.UpdateResourceRequest, resp *tfsdk.UpdateResourceResponse) {
+	resp.Diagnostics.AddError("Unable to update API Key", "Updating API Keys is not currently supported. Delete and recreate the provider.")
+	return
+}
+
+// Revoke API Key
+func (r resourceApiKey) Delete(ctx context.Context, req tfsdk.DeleteResourceRequest, resp *tfsdk.DeleteResourceResponse) {
+	var state ApiKey
+	getIDsFromApiKeyState(ctx, req.State, &state)
+	accountId := state.AccountID.Value
+	apiKeyId := state.ApiKeyID.Value
+
+	apiClient := r.p.client
+
+	response, err := apiClient.AuthApi.RevokeApiKey(context.Background(), accountId, apiKeyId).Execute()
+	if err != nil {
+		errMsg := getErrorMessage(response, err)
+		resp.Diagnostics.AddError("Unable to revoke the API Key ", errMsg)
+		return
+	}
+
+	resp.State.RemoveResource(ctx)
+
+}
+
+// Import API Key
+func (r resourceApiKey) ImportState(ctx context.Context, req tfsdk.ImportResourceStateRequest, resp *tfsdk.ImportResourceStateResponse) {
+	// Save the import identifier in the id attribute
+	tfsdk.ResourceImportStatePassthroughID(ctx, path.Root("id"), req, resp)
```

Comment:
What is the import identifier ?

Reply:
Uhh not sure.. it was there in all the resources
No idea where its used..

Reply:
@shinji62 The ImportState function is used to import state from a remote resource https://developer.hashicorp.com/terraform/plugin/framework/resources/import#define-resource-importstate-method, our provider doesn't support it yet. We've just let the default code be for all the resources. We still need to test and document the import command.

Reply:
@posriniv Not totally true, I did test and implement import for VPC, NAL and PSE (merged)